naga/src/lib.rs

```diff
@@ -2017,6 +2019,52 @@ pub enum Statement {
         /// [`SHADER_INT64_ATOMIC_ALL_OPS`]: crate::valid::Capabilities::SHADER_INT64_ATOMIC_ALL_OPS
         result: Option<Handle<Expression>>,
     },
+    /// Performs an atomic operation on a texel value of an image.
+    ///
+    /// Doing atomics on images with mipmaps is not supported, so there is no
+    /// `level` operand.
+    ///
+    /// This statement is a barrier for any operations on the corresponding
+    /// [`Expression::GlobalVariable`] for this image.
```

Comment:
This is the case on metal too? If so that's great

Reply:
rereading the spec, im not sure. 6.15.1 says it essentially depends on the memory ordering specified https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf